packages/legend-studio/src/components/editor/edit-panel/project-configuration-editor/ProjectConfigurationEditor.tsx

```diff
@@ -230,7 +230,20 @@ const ProjectDependencyEditor = observer(
     // version
     const version = projectDependency.versionId;
     const versions = selectedProject?.versions ?? [];
-    const versionOptions = versions.map((v) => ({ value: v, label: v }));
+    const versionOptions = versions
```

Comment:
could we look into what options we have with `npm`? This doesn't seem like the proper comparison to me. Also, if we cannot go with 3rd party library, perhaps, let's go with an utility function that we can unit test, I would be more confident then.

That being said, why don't we use what the backend return and just revert it, don't they sorta ensure the order for us in some way?

Reply:
this is what backend returns.
<img width="494" alt="Screen Shot 2022-03-04 at 3 35 46 PM" src="https://user-images.githubusercontent.com/73408381/156837870-ac33ca1d-b9f0-4a21-9521-c65af17ad9db.png">
e.g.
1.0 -> 11.2.0 -> 12.0.0 ->  2.1.0

Reply:
funny how `2.0.0` shows up in the end there.
Anyway, I think we can use this https://www.npmjs.com/package/compare-versions

Reply:
Or we can do this

https://stackoverflow.com/questions/55466274/simplify-semver-version-compare-logic/55466325#55466325

https://github.com/omichelsen/compare-versions/issues/45

That's not bad. We can publish this as part of utilities in `DepotUtils`, we can set a note there linking to this library and this issue.

I can merge this and do that myself if you'd rather me doing that. Let me know!